### PR TITLE
add delete_branch_on_merge and is_template options

### PR DESCRIPTION
**Pull request by soerenmartius**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -100,6 +101,12 @@ variable "allow_rebase_merge" {
   default     = null
 }
 
+variable "delete_branch_on_merge" {
+  description = "Whether or not to delete the merged branch after merging a pull request. (Default: false)"
+  type        = bool
+  default     = false
```
Comment (mariux) [inline variables.tf:107]: should be `null`
```suggestion
  default     = null
```

File: variables.tf [hunk 2]
```diff
@@ -118,6 +125,12 @@ variable "gitignore_template" {
   default     = null
 }
 
+variable "is_template" {
+  description = "Whether or not to tell GitHub that this is a template repository. ( Default: false)"
+  type        = bool
+  default     = false
```
Comment (mariux) [inline variables.tf:131]: should be `null` for defaults
```suggestion
  default     = null
```

Comment (soerenmartius) [reply variables.tf]: fixed in 84ebec5

File: variables.tf (hunk 1 above)
Comment (soerenmartius) [reply variables.tf]: fixed in 84ebec5

**Comment by soerenmartius** [comment]
Please review once again @mariux

**Comment by mariux** [comment]
@soerenmartius this catually can be considered a breaking change as it will revert all manual set delete-on-branch.. 

did you consider to make this a default to true setting?

**Comment by mariux** [comment]
how does the resource handle it when defaulting is set to `null`? will it keep manual changes or disable them?

**Pull request merged**